src/models/Store.js

```diff
@@ -0,0 +1,21 @@
+import { observable } from "mobx";
+
+import User from './User';
+
+export const checked = 'checked';
+export const empty = 'empty';
+export const loading = 'loading';
+
+class Store {
+  @observable status = 'empty';
+  @observable user;
+
+  async loadUser() {
+    this.status = 'loading';
```

Comment:
I hate using this magic strings but i haven't quiet figure out how to use them from the declaration above. Leaving it for now just because i wanna move on.

Reply:
no concept of enums or the like?

Reply:
no :( it makes me sad. I think this can be better for sure. I need to think about how to make this state transition elegant, but this does the job for now.

Reply:
lame...